### DISCUSSION TITLE
Kakao login with signInWithIdToken

### DISCUSSION THREAD

**Discussion by leeeeeoy** [supabase/supabase#23368] [Questions]
Hello, I have a question.
I'd like to log in to Kakao using ID Token. Is there a way? The existing package only allows Google and Apple, so I tried to log in by modifying the package, but I encountered an error called 'unacceptable audio in id_token'. It would be helpful if you could tell me a solution or document. Thank you always for providing such a good project.

**Comment by nectarine** [comment]
Thanks to @MiryangJung, Supabase already accept Kakao OIDC on the supabase.auth side (see [PR #1381](https://github.com/supabase/auth/pull/1381)).

If you retrieved an id_token using a REST API KEY, you must enter the REST API KEY in the Supabase dashboard. In this case, you cannot use a native Kakao login (Android or iOS) with the NATIVE APP KEY. (You need to use a `signInWithOAuth`)

If you retrieved an id_token using the NATIVE APP KEY through the native login process, you must enter the NATIVE APP KEY in the REST API KEY field in the Auth Provider dashboard (ignore the REST API KEY label, as it is just a ClientId in supabase.auth). Additionally, you may need to modify the client library to pass the validation, as done in [this PR for supabase_flutter](https://github.com/supabase/supabase-flutter/pull/922).
However, it blocks REST API KEY, so you cannot provide sign in with the web application.
(in the REST API KEY scenario above, despite bad ux, the user at least can sign in with the mobile browser)

The main reason for the error (unacceptable aud in id_token) is that the id_token contains an `aud` with the value of the API key used for retrieving the id_token. This `aud` should match the REST API KEY value (again, ignore the label) in the Supabase dashboard.

Therefore, both server and client tasks are required to fully handle `signInWithIdToken` using Kakao OIDC for both REST and native app sides.

Server side:
- The Supabase studio (dashboard) needs an additional field for the NATIVE APP KEY in the Kakao section.
- supabase.auth should handle this configuration value for aud matching.

Client side:
- As seen with supabase_flutter, for any Supabase client library, if there is a validation inside signInWithIdToken() blocking OAuthProvider.kakao, it must be fixed.

I plan to fix server-side issues this month. Meanwhile, it is highly encouraged that you fix the client library you are using in advance(if it is not Flutter).

**Reply by leeeeeoy** [reply]
Thank you for your kind reply, and I will close the discussion.
